pkg/proctree/nodes/process.go

```diff
@@ -0,0 +1,239 @@
+package nodes
```

Comment:
nit: I think that considering the content, this is just the `process` package (unless you think there will be other types of nodes in the future?).

Reply:
I wanted to make sure future expansions would be encapsulated and private to a pkg so locks aren't exposed in between the packages.

Reply:
Not sure I get how that's related? There's no package level lock as far as I can tell. The package and file content so far only relates to the instantiation and definition of a process struct, so the package name would fit better imo.

Reply:
Ahh nm I thought u meant something else. Alright.